pkg/statistics/handle/ddl/reorganize_partition.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"github.com/pingcap/tidb/pkg/statistics/handle/util"
+)
+
+func (h *ddlHandlerImpl) onReorganizePartitions(t *util.DDLEvent) error {
+	globalTableInfo,
+		addedPartInfo,
+		droppedPartitionInfo := t.GetReorganizePartitionInfo()
+	// Avoid updating global stats as the data remains unchanged.
+	// For new partitions, it's crucial to correctly insert the count and modify count correctly.
+	// However, this is challenging due to the need to know the count of the new partitions.
+	// Given that a partition can be split into two, determining the count of the new partitions is so hard.
+	// It's acceptable to not update it immediately,
+	// as the new partitions will be analyzed shortly due to the absence of statistics for them.
+	// Therefore, the auto-analyze worker will handle them in the near future.
+	for _, def := range addedPartInfo.Definitions {
+		if err := h.statsWriter.InsertTableStats2KV(globalTableInfo, def.ID); err != nil {
```

Comment:
Can we do it with batch mode? 

```
insert into mysql.stats_meta xxx values (data1),(data2) 
```

Reply:
It is a little difficult because the current API does not support it. However, I don't think we will need to reorganize too many partitions, so I guess it's okay.